sdk/core/Azure.Core/tests/TestEnvironmentTests.cs

```diff
@@ -85,27 +85,25 @@ public void ReadingNonRecordedValueInCtorWorks(RecordedTestMode mode)
         [TestCase(RecordedTestMode.Playback)]
         [TestCase(RecordedTestMode.Record)]
         [TestCase(RecordedTestMode.None)]
-        public void ReadingRecordedValueInCtorWorksForSamples(RecordedTestMode mode)
+        public void ReadingRecordedValueInCtorDoesNotWorkForSamples(RecordedTestMode mode)
```

Comment:
Hm, I now remember. We tried to have the validation in recorded tests but skip it in other contexts. I'm afraid things would start exploding in Live tests if we tighten it. 

What do you think about adding a `DisallowRecordedVariables` property to `TestEnvironment`, setting it to true in the RecordedTestBase ctor and to false in StartRecording?

Reply:
Maybe we should change https://github.com/Azure/azure-sdk-for-net/blob/c9eb735fcebe56c948d9c6804e4b99f53636ebf1/sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs#L252-L255
to
```
if (Mode == Live) {
  return value;
}
```
?

Reply:
I think the problem is even earlier. 

https://github.com/Azure/azure-sdk-for-net/blob/c9eb735fcebe56c948d9c6804e4b99f53636ebf1/sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs#L245-L248

Now that the default is playback instead of "null" the control flow falls into the first if.

Reply:
OR instead of validating  we if-else in _recording presence and emit warning to TestContext.Error until usage of this api is fixed across the board.

Reply:
I don't think it needs to be fixed.

The fix would be adding another copy of GetVariable as you had to do with identity. Not sure if that helps much or adds noise.